### PR TITLE
Pytest for 1195

### PR DESCRIPTION
**Pull request by neilCrosswaite**
Git diff has squiffed up.
Follow the two commits for clear understanding.

Made pytests for the new convert-to-realization CLI

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: lib/improver/tests/acceptance/test_convert_to_realizations.py [hunk 1]
```diff
@@ -28,24 +28,46 @@
...
+    input_dir = kgo_dir / "../basic"
+    input_path = input_dir / "input.nc"
+
+    output_path = tmp_path / "output.nc"
+
+    args = [input_path,
+            "--no-of-realizations", "12",
+            "--output", output_path]
+    run_cli(args)
+    acc.compare(output_path, kgo_path)
```
Comment (benfitzpatrick) [inline lib/improver/tests/acceptance/test_convert_to_realizations.py]: Codacy Issue found: [Final newline missing](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4656787)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- lib/improver/tests/acceptance/test_convert_to_realizations.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4656787&bid=15595720)

File: lib/improver/tests/acceptance/test_convert_to_realizations.py (hunk 1 above)
Comment (neilCrosswaite) [inline lib/improver/tests/acceptance/test_convert_to_realizations.py]:
```suggestion
    acc.compare(output_path, kgo_path)
    
```

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- lib/improver/tests/acceptance/test_convert_to_realizations.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4656787&bid=15595720)

File: lib/improver/tests/acceptance/test_convert_to_realizations.py [hunk 2]
```diff
@@ -28,24 +28,47 @@
...
+
```
Comment (benfitzpatrick) [inline lib/improver/tests/acceptance/test_convert_to_realizations.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4656787)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- lib/improver/tests/acceptance/test_convert_to_realizations.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4656787&bid=15595720)

File: lib/improver/tests/acceptance/test_convert_to_realizations.py (hunk 1 above)
Comment (benfitzpatrick) [inline lib/improver/tests/acceptance/test_convert_to_realizations.py]: Codacy Issue found: [Final newline missing](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4656787)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- lib/improver/tests/acceptance/test_convert_to_realizations.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4656787&bid=15595720)

File: lib/improver/tests/acceptance/test_convert_to_realizations.py [hunk 3]
```diff
@@ -28,24 +28,47 @@
...
```
Comment (benfitzpatrick) [inline lib/improver/tests/acceptance/test_convert_to_realizations.py]: Codacy Issue found: [Trailing newlines](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4656787)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- lib/improver/tests/acceptance/test_convert_to_realizations.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4656787&bid=15595720)

File: lib/improver/tests/acceptance/test_convert_to_realizations.py (hunk 1 above)
Comment (benfitzpatrick) [inline lib/improver/tests/acceptance/test_convert_to_realizations.py]: Codacy Issue found: [Final newline missing](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4656787)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- lib/improver/tests/acceptance/test_convert_to_realizations.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4656787&bid=15595720)

File: lib/improver/tests/acceptance/test_convert_to_realizations.py (hunk 1 above)
Comment (benfitzpatrick) [inline lib/improver/tests/acceptance/test_convert_to_realizations.py]: Codacy Issue found: [Final newline missing](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4656787)

Comment (neilCrosswaite) [reply lib/improver/tests/acceptance/test_convert_to_realizations.py]: But there is!

File: lib/improver/tests/acceptance/test_convert_to_realizations.py [hunk 4]
```diff
@@ -28,24 +28,46 @@
...
+CLI = acc.cli_name_with_dashes(__file__)
+run_cli = acc.run_cli(CLI)
 
-  improver_check_recreate_kgo "output.nc" $KGO
 
-  # Run nccmp to compare the output and kgo.
-  improver_compare_output "$TEST_DIR/output.nc" \
-      "$IMPROVER_ACC_TEST_DIR/$KGO"
-}
+def test_percentiles(tmp_path):
```
Comment (tjtg) [inline lib/improver/tests/acceptance/test_convert_to_realizations.py]: This test takes ~10 seconds to run - please mark it as slow with `@pytest.mark.slow`.

File: lib/improver/tests/acceptance/test_convert_to_realizations.py [hunk 5]
```diff
@@ -28,24 +28,46 @@
...
+    input_path = kgo_dir / "multiple_percentiles_wind_cube.nc"
+    output_path = tmp_path / "output.nc"
+    args = [input_path,
+            "--no-of-realizations", "12",
+            "--output", output_path]
+    run_cli(args)
+    acc.compare(output_path, kgo_path)
+
+
+def test_probabilities(tmp_path):
```
Comment (tjtg) [inline lib/improver/tests/acceptance/test_convert_to_realizations.py]: This test takes ~6 seconds to run - please mark it as slow.

File: lib/improver/tests/acceptance/test_convert_to_realizations.py [hunk 6]
```diff
@@ -28,24 +28,46 @@
...
+    """Test basic probabilities to realization conversion"""
+    kgo_dir = (acc.kgo_root() /
+               "probabilities-to-realizations/12_realizations")
+    kgo_path = kgo_dir / "kgo.nc"
+    input_dir = kgo_dir / "../basic"
+    input_path = input_dir / "input.nc"
+
```
Comment (tjtg) [inline lib/improver/tests/acceptance/test_convert_to_realizations.py]: The two tests here have blank lines in different locations - suggest choosing one way to do it so these are consistent.

File: lib/improver/tests/acceptance/test_convert_to_realizations.py (hunk 1 above)
Comment (tjtg) [reply lib/improver/tests/acceptance/test_convert_to_realizations.py]: The line break at the end of file looks fine to me when I checked out the branch. I'm not sure why codacy is wrong here.

File: lib/improver/tests/acceptance/test_convert_to_realizations.py (hunk 6 above)
Comment (neilCrosswaite) [reply lib/improver/tests/acceptance/test_convert_to_realizations.py]: blank line either side of output.

File: lib/improver/tests/acceptance/test_convert_to_realizations.py (hunk 5 above)
Comment (neilCrosswaite) [reply lib/improver/tests/acceptance/test_convert_to_realizations.py]: Done

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- lib/improver/tests/acceptance/test_convert_to_realizations.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4656787&bid=15595720)

File: lib/improver/tests/acceptance/test_convert_to_realizations.py [hunk 7]
```diff
@@ -28,24 +28,49 @@
...
+@pytest.mark.slow
+def test_percentiles(tmp_path):
+    """Test basic percentile to realization conversion"""
+    kgo_dir = (acc.kgo_root() /
+               "percentiles-to-realizations/percentiles_rebadging")
+    kgo_path = kgo_dir / "kgo.nc"
+    input_path = kgo_dir / "multiple_percentiles_wind_cube.nc"
+    
+    output_path = tmp_path / "output.nc"
+
```
Comment (benfitzpatrick) [inline lib/improver/tests/acceptance/test_convert_to_realizations.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4656787)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/improver/tests/acceptance/test_convert_to_realizations.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4656787&bid=15595720)

**Pull request merged**